python/ray/tests/test_tempfile.py

```diff
@@ -38,27 +38,27 @@ def test_conn_cluster():
         "temp_dir must not be provided.")
 
 
-def test_tempdir():
-    shutil.rmtree("/tmp/ray", ignore_errors=True)
+def test_tempdir(shutdown_only):
+    try_remove_directory("/tmp/ray")
     ray.init(temp_dir="/tmp/i_am_a_temp_dir")
     assert os.path.exists(
         "/tmp/i_am_a_temp_dir"), "Specified temp dir not found."
     assert not os.path.exists("/tmp/ray"), "Default temp dir should not exist."
     ray.shutdown()
```

Comment:
We don't need this and `shutdown_only`, right?

Reply:
There are other instances of this in this PR as well.

Reply:
I think `shutdown_only` is necessary for most Ray tests because in case the test fails, it will not clean up correctly without `shutdown_only`.

Reply:
I mean we can remove `ray.shutdown()` if we use the `shutdown_only` fixture correct?

Reply:
@suquark I think @devin-petersohn is right that we are calling `ray.shutdown()` twice here.

Reply:
My concern is that when a test fails and we do not use `shutdown_only` fixture, it will also affect the remaining tests. So I think `shutdown_only` should be used for most test. In this test, we have to call `ray.shutdown()` first to close related file handles, so it cannot be replaced by `shutdown_only`.